### PR TITLE
Workbranch

### PR DESCRIPTION
**Pull request by harshwalia36**

### CODE REVIEW

File: src/api/controllers/signup.js [hunk 1]
```diff
@@ -11,9 +11,18 @@ const handleSignup = async (req, res, db) => {
   if (errors.length) res.status(400).send(errors);
   else {
     const hash = await bcrypt.hash(password, 10);
-    await db.collection('users').insertOne({ name, email, phone, hash });
+    await db
+      .collection('users')
+      .insertOne({ name, email, phone, hash, isAdmin: false, isVerified: false });
```
Comment (Mallik813) [inline src/api/controllers/signup.js]: make the field as `isAdminVerified` let isVerified be for email verification.
It'll be a confusion later.

File: src/api/controllers/posts.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+module.exports = async (req, res, db) => {
+  const { title, text } = req.body;
+  const user = await db.collection('users').findOne({ email: req.user.email });
+  const ID = user._id;
+  await db.collection('posts').insertOne({ ID, title, text });
+  const post = await db.collection('posts').findOne({ ID });
```
[the remark this answers is not in the document]
Comment (Mallik813) [reply src/api/controllers/posts.js]: But before finding the id of the post you are finding the post which was made by the user. A single user can make any number of posts right. You may not always get the recent post through the query from line 6.

File: src/api/index.js [hunk 3]
```diff
@@ -21,13 +23,35 @@ MongoClient.connect(url, { useUnifiedTopology: true }, (err, client) => {
...
-    signup(req, res, db);
+    signup.handleSignup(req, res, db);
   });
   router.post('/login', (req, res) => {
     login(req, res, db);
   });
+  router.get('/admin', userAuth, verifyAdmin(db), async (req, res) => {
+    const userToBeVerified = await db
+      .collection('admins')
+      .find({})
```
Comment (Mallik813) [inline src/api/index.js]: This just returns the ids of users. You should not return this. You should make another query to users collection and get the users' documents.

File: src/api/controllers/posts.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (harshwalia36) [reply src/api/controllers/posts.js]: OK, I got it.

File: src/api/index.js [hunk 4]
```diff
@@ -21,13 +22,34 @@ MongoClient.connect(url, { useUnifiedTopology: true }, (err, client) => {
...
-    signup(req, res, db);
+    signup.handleSignup(req, res, db);
   });
   router.post('/login', (req, res) => {
     login(req, res, db);
   });
+  router.get('/admin', userAuth, async (req, res) => {
+    const { email } = req.body;
+    const authCheck = await db.collection('users').findOne({ email });
+    const { isAdmin } = authCheck;
```
[the remark this answers is not in the document]
Comment (harshwalia36) [reply src/api/index.js]: resolved

**Pull request merged**